build-coordinator/src/main/java/org/jboss/pnc/coordinator/maintenance/DefaultRemoteBuildsCleaner.java

```diff
@@ -144,7 +144,7 @@ private Result deleteBuildsFromIndy(BuildRecord buildRecord, String authToken) {
         Indy indy = indyFactory.get(authToken);
         try {
             IndyStoresClientModule indyStores = indy.stores();
-            if (pkgKey != null) {
+            if (buildRecord.getStatus().completedSuccessfully() && pkgKey != null) {
```

Comment:
This would cause that the uploaded content and the hosted repo cotaining it would not be deleted for failed builds. Please wrap only the deletion from temporary builds (following 6 lines) into the condition. The lines below `delete the content` still have to be executed.

Reply:
Thank you very much, you are absolutely correct. I've not noticed that there are 2 deletions happening in the `if` block.